### ISSUE TITLE
Could not initialize random number generator

### ISSUE THREAD

**Issue by the-impaler** [docker-library/wordpress#630]
**Description**

Container starts and instantly stops with the following Error: `[Thu Aug 19 09:19:01.413647 2021] [:crit] [pid 1] (38)Function not implemented: AH00141: Could not initialize random number generator` - it does this whether the container is freshly created or restarted etc.

I'm aware that this error is thrown when running a kernel older than 3.17 as this is where where getrandom was introduced. I am however unable to upgrade to a newer kernel. Is there an image tag which does not have this random number generator, or could you consider a more graceful fallback to an older option where the newer getrandom command is not available? Is there any other workaround available? This was working in the version immediately prior to last nights update, but I cannot see a valid tag for that older version.

Let me know if I can provide additional information and thanks in advance.

**Steps to reproduce the issue:**

1. Use a Docker Host with Kernel 3.10.102
2. Run Wordpress container using latest tag
3. Try to start it

**Describe the results you received:**

It stops with the error above

**Version**

- Output of docker version:

```
Client:
 Version:           20.10.3
 API version:       1.41
 Go version:        go1.15.6
 Git commit:        b35e731
 Built:             Fri Jun 18 08:25:45 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.3
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.15.6
  Git commit:       e7f7c95
  Built:            Fri Jun 18 08:26:10 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        b1dc45ec561bd867c4805eee786caab7cc83acae
 runc:
  Version:          v1.0.0-rc93
  GitCommit:        89783e1862a2cc04647ab15b6e88a0af3d66fac3
 docker-init:
  Version:          0.19.0
  GitCommit:        12b6a20
```

- Output of docker info:

```
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 32
  Running: 23
  Paused: 0
  Stopped: 9
 Images: 46
 Server Version: 20.10.3
 Storage Driver: aufs
  Root Dir: /volume1/@docker/aufs
  Backing Filesystem: extfs
  Dirs: 0
  Dirperm1 Supported: true
 Logging Driver: db
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan null overlay
 Kernel Version: 3.10.102
 Operating System:  (containerized)
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.59GiB
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:

```

- Output of docker-compose version (if applicable):

`docker-compose version 1.28.5, build 1bbbad71
docker-py version: 4.4.4
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019`

- Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):

Host is running Synology DSM 6.1.7 and cannot be upgraded

**Comment by FlamingFlowair** [comment]
Hello, I'm facing this issue also on a DS916+ which kernel version is stuck to 3.10.108
DSM7 update does not matter as it seems Synology never make any kernel major version update on its products.

**Comment by wglambert** [comment]
Yeah looks like this is related to using kernel version 3.16 or below
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=978045

**Comment by the-impaler** [comment]
Thanks for tracking that down. I presume this is a "can't fix" then from reading the Debian bugtracker it looks like they just see kernels in the 3.X range as too old to continue to support, and that is outside of your control?

Presume you also wouldn't be willing to create an image with an older version of Apache without this dependency?

**Comment by FlamingFlowair** [comment]
One solution that could work out for me is to switch to another docker image.
Either php-fpm + another container for Nginx
Or using another docker image with Wordpress under Nginx like https://github.com/bitnami/bitnami-docker-wordpress-nginx

**Comment by yosifkit** [comment]
You could build your own by using our WordPress Dockerfiles and pin to the Debian Buster based PHP images by changing the `FROM` line:

```diff
-FROM php:8.0-apache
+FROM php:8.0-apache-buster
```
